docs/proposal-link-definition-and-disambiguation.md

```diff
@@ -0,0 +1,51 @@
+# Proposal: Link Definition and Disambiguation
+
+## Current status
+
+- We generate link ref defs from wiki-links inline
+- Wiki links are expected to be the full source of truth, but we don't have enough information to conclusively disambiguate
+
+## Solution
+
+- Don't wipe out link ref defs. Only add/remove as needed. This can be our metadata for disambiguation.
+
+Let's say we have two documents in the workspace with the same title "Some Title"
+
+- We would have autocompletion base on title of the document
+  - This generates a link (typically a title, but could also have e.g. `:2` to disambiguate multiple links in the same file)
```

Comment:
Personally I would prefer the local links to use style [[foo/bar/my-title]] - would this be configurable?

Reply:
@jojanaho, to clarify, do you mean you prefer `foo/bar/my-title` for just the disambiguated links in case of title conflicts, or always?

It could be configurable, since it's an arbitrary label. Options could include:
- file-name.md
- Title Cased File Name
- Document Title
- Arbitrary Label you write yourself

Reply:
Always, since in my case it would improve readability - e.g [[book-summaries/war-and-peace]]

Reply:
And to be exact, we have the freedom for arbitrary labels only when link ref defs are in use :)